src/XMakeTasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1535,15 +1520,56 @@ Copyright (C) Microsoft Corporation. All rights reserved.
           there is also a .NETFramework implementation.
 
        2. As an escape hatch for cases where the compatibility check performed by 
-          GetTargetFrameworkProperties is faulty.
+          GetReferenceProperties is faulty.
     -->
     <ItemGroup>
       <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.SetTargetFramework)' != ''">
-        <SkipGetTargetFrameworkProperties>true</SkipGetTargetFrameworkProperties>
+        <SkipGetReferenceProperties>true</SkipGetReferenceProperties>
       </_MSBuildProjectReferenceExistent>
     </ItemGroup>
 
     <!--
+      Honor SkipGetReferenceProperties=true metadata on project references
+      to mean that the project reference is known not to target multiple frameworks
+      and the mechanism here for selecting the best one can be skipped as an optimization.
+    -->
+    <MSBuild
+        Projects="@(_MSBuildProjectReferenceExistent)"
+        Targets="ShouldQueryForProperties"
+        BuildInParallel="$(BuildInParallel)"
+        Properties="%(_MSBuildProjectReferenceExistent.SetConfiguration); %(_MSBuildProjectReferenceExistent.SetPlatform)"
+        ContinueOnError="!$(BuildingProject)"
+        RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove);TargetFramework;RuntimeIdentifier"
```

Comment:
It would be nice as escape hatch if the set of properties removed from ShouldQueryForProperties were configurable and defaulted TargetFramework;RuntimeIdentifier. I'm thinking that another global property might come along and bite us later on and an escape hatch here could help. This can be done/tracked separately though.